### PR TITLE
add --alldeps options

### PR DESCRIPTION
**Pull request by oliverkurth**
This adds the `--alldeps` option. This can be used together with the `--downloadonly` option to download all dependencies regardless whether they are installed or not. This is useful for example to create a repository for another host behind a firewall that cannot access the repositories directly.

`reposync` could be used alternatively, but this limits the download to the package(s) to be installed and their requirements.

Example usage:
```
tdnf install --alldeps --downloadonly --downloaddir=/tmp/lsof lsof

Installing:
pcre-libs                          x86_64                    8.44-3.ph4                         photon-updates            480.89k                     179.22k
libselinux                         x86_64                    3.3-1.ph4                          photon-updates            170.36k                      78.81k
grep                               x86_64                    3.4-1.ph4                          photon-release            229.47k                     121.20k
gmp                                x86_64                    6.2.0-1.ph4                        photon-release            532.45k                     266.22k
coreutils-selinux                  x86_64                    9.1-1.ph4                          photon-updates              6.97M                       1.35M
libgcc                             x86_64                    10.2.0-1.ph4                       photon-release            100.82k                      54.88k
openssl                            x86_64                    3.0.3-3.ph4                        photon-updates              6.17M                       2.20M
krb5                               x86_64                    1.17-7.ph4                         photon-updates              3.34M                       1.17M
ncurses-libs                       x86_64                    6.2-6.ph4                          photon-updates            625.83k                     253.90k
e2fsprogs-libs                     x86_64                    1.45.6-2.ph4                       photon-release             58.37k                      25.40k
libtirpc                           x86_64                    1.2.6-2.ph4                        photon-updates            193.56k                      93.03k
readline                           x86_64                    7.0-3.ph4                          photon-release            371.60k                     144.09k
filesystem                         x86_64                    1.1-4.ph4                          photon-release              7.19k                      20.85k
bash                               x86_64                    5.0-2.ph4                          photon-release              1.82M                     552.06k
glibc                              x86_64                    2.32-11.ph4                        photon-updates              8.72M                       2.62M
lsof                               x86_64                    4.91-1.ph4                         photon-release            198.47k                     115.62k

Total installed size:  29.93M
Total download size:   9.20M
tdnf will only download packages needed for the transaction
Is this ok [y/N]: 
```

After creating a repo:
```
createrepo_c /tmp/lsof/
```

on another host, one could then run (assuming the download directory is shared or copied to the same location):
```
tdnf install --repofrompath=lsof,/tmp/lsof lsof

Installing:
libtirpc                           x86_64                    1.2.6-2.ph4                        lsof                      193.56k                      93.03k
lsof                               x86_64                    4.91-1.ph4                         lsof                      198.47k                     115.62k

Total installed size: 392.03k
Total download size: 208.65k
Is this ok [y/N]: 
```

This change borrows the idea and some code snippets from https://github.com/microsoft/CBL-Mariner/blob/2.0/SPECS/tdnf/tdnf-add-download-command.patch but is much simpler and uses much less code while implementing the same functionality.

### CODE REVIEW

File: solv/tdnfpool.c [hunk 1]
```diff
@@ -67,7 +67,8 @@ uint32_t
 SolvInitSack(
     PSolvSack *ppSack,
     const char* pszCacheDir,
-    const char* pszRootDir
+    const char* pszRootDir,
+    int nSkipInstalledRpms
```
Comment (tapakund) [inline solv/tdnfpool.c]: shall we rename nSkipInstalledRpms to nInstallAllDeps as it looks little confusing? What are your thoughts?

I understand concept is same, but it feels easy to read and understand code with this name. I am open for any new suggestions as well.

**Review by tapakund — COMMENTED** [review]
Looks good, just one discussion on naming an argument.

File: solv/tdnfpool.c (hunk 1 above)
Comment (oliverkurth) [reply solv/tdnfpool.c]: Sure, I can do that.

**Review by tapakund — APPROVED** [review]
changes looks good.

**Pull request merged**